### PR TITLE
build: Cleaned up and consolidated msbuild files (no code changes)

### PR DESCRIPTION
**Pull request by sipsorcery**
This PR does the following:

 - Big cleanup of the msbuild project files and consolidation of common build parameters into a single common include,
 - Gets ready for switch to building with Visual Studio 2019 and Platform Toolset v142,
 - Will allow me to rebase #15529 to include the Qt Projects in the msbuild config.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
The linter complains about white space. Idk, are you using the windows-new-line-encoding?

**Comment by MarcoFalke** [comment]
Any suggestions on who is qualified to review this?

**Comment by sipsorcery** [comment]
> The linter complains about white space. Idk, are you using the windows-new-line-encoding?

@MarcoFalke not intentionally. 

Is there a trick to running the linter script locally? I tried `test/lint/lint-all.sh` from the root of my git repo but it didn't do anything useful.

**Comment by sipsorcery** [comment]
> Any suggestions on who is qualified to review this?

@ken2812221 @NicolasDorier ?

**Comment by practicalswift** [comment]
Concept ACK: nice cleanup!

**Comment by fanquake** [comment]
Concept ACK for such a large reduction in Windows specific build config. Is this a result of their build system improving, or you figuring out a better way to make this work?

> Gets ready for switch to building with Visual Studio 2019 and Platform Toolset v142

Does this drop compatibility with older versions of Visual Studio?

I will review in so far as spinning up a Windows VM and making sure the project opens / builds inside Visual Studio.

File: build_msvc/common.init.vcxproj [hunk 1]
```diff
@@ -1,14 +1,108 @@
...
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-      <PropertyGroup Condition="'$(WindowsTargetPlatformVersion)'=='' and !Exists('$(WindowsSdkDir)\DesignTime\CommonConfiguration\Neutral\Windows.props')">
-        <WindowsTargetPlatformVersion_10 Condition="'$(WindowsTargetPlatformVersion_10)' == ''">$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\Windows\v10.0@ProductVersion)</WindowsTargetPlatformVersion_10>
-        <WindowsTargetPlatformVersion_10 Condition="'$(WindowsTargetPlatformVersion_10)' == ''">$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0@ProductVersion)</WindowsTargetPlatformVersion_10>
-        <!-- Sometimes the version in the registry has to .0 suffix, and sometimes it doesn't. Check and add it -->
-        <WindowsTargetPlatformVersion_10 Condition="'$(WindowsTargetPlatformVersion_10)' != '' and !$(WindowsTargetPlatformVersion_10.EndsWith('.0'))">$(WindowsTargetPlatformVersion_10).0</WindowsTargetPlatformVersion_10>
-
-        <WindowsTargetPlatformVersion>$(WindowsTargetPlatformVersion_10)</WindowsTargetPlatformVersion>
-    </PropertyGroup>
```
Comment (NicolasDorier) [inline build_msvc/common.init.vcxproj]: Why you remove this?
I found out that it is the best way to get bitcoin to build on several versions of visual studio.

**Comment by NicolasDorier** [comment]
Concept ACK. Will test. Though you removed the check in registry to make bitcoin build on wider configs.

I noticed that visual studio or msbuild via command line would get quite capricious if the exact right version of SDK is not installed.

File: build_msvc/common.init.vcxproj (hunk 1 above)
Comment (sipsorcery) [reply build_msvc/common.init.vcxproj]: Restored.

**Comment by sipsorcery** [comment]
> Is this a result of their build system improving, or you figuring out a better way to make this work?

It's a result of a lot of optimising combined with trial and error on the .vcxproj files. Typically the msbuild project files are left alone and managed by Visual Studio. There's no need to edit them manually. That doesn't make them very maintainable when every change needs to be conscientiously reviewed.

> Does this drop compatibility with older versions of Visual Studio?

The build still runs successfully on appveyor with the Visual Studio 2017 image. I suspect the build should work correctly on VS 2012 and 2015 but I haven't tested.

**Comment by sipsorcery** [comment]
> Is this a result of their build system improving, or you figuring out a better way to make this work?

It's a result of a lot of optimising combined with trial and error on the .vcxproj files. Typically the msbuild project files are left alone and managed by Visual Studio. There's no need to edit them manually. That doesn't make them very maintainable when every change needs to be conscientiously reviewed.

> Does this drop compatibility with older versions of Visual Studio?

The build still runs successfully on appveyor with the Visual Studio 2017 image. I suspect the build should work correctly on VS 2012 and 2015 but I haven't tested.

**Comment by fanquake** [comment]
ACK 750d1bc7df5ec6505a8fb92dabface887c23817c

I setup a new Windows 10 VM using [this guide](https://github.com/fanquake/core-review/blob/master/windows.md), used this branch + installed `rapidcheck` (#16241) and successfully built a working `bitcoind` and all tests seem to be passing.

windows


> The build still runs successfully on appveyor with the Visual Studio 2017 image. I suspect the build should work correctly on VS 2012 and 2015 but I haven't tested.

Ok. That sounds fine, but will defer to the other Windows contributors as to what is acceptable.

**Comment by NicolasDorier** [comment]
tACK 750d1bc . Good to see the boilerplate go away.

**Review by NicolasDorier — APPROVED** [review]

**Pull request merged**